extensions/amp-story/1.0/amp-story.js

```diff
@@ -1350,26 +1360,37 @@ export class AmpStory extends AMP.BaseElement {
       return Promise.resolve();
     }
 
-    // Block until the subscription state gets resolved.
+    // Block until the subscriptions page index gets resolved.
     const subscriptionsState = this.storeService_.get(
       StateProperty.SUBSCRIPTIONS_STATE
     );
     if (
-      pageIndex >= PAYWALL_PAGE_INDEX &&
+      subscriptionsState !== SubscriptionsState.DISABLED &&
+      this.subscriptionsPageIndex_ === -1
+    ) {
+      return this.blockOnPendingSubscriptionsPageIndex_(
```

Comment:
So two states should be blocked and waited at different times. This is because sub state is a network request and takes much longer than getting the sub index from `amp-story-subscriptions`. This is why the sub index is blocked on the initial layout switchTo and the sub state is only blocked when user is trying to navigate to the page after sub index.